beacon-chain/db/kv/attestations.go

```diff
@@ -159,33 +153,21 @@ func (k *Store) SaveAttestation(ctx context.Context, att *ethpb.Attestation) err
 func (k *Store) SaveAttestations(ctx context.Context, atts []*ethpb.Attestation) error {
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.SaveAttestations")
 	defer span.End()
-	encodedValues := make([][]byte, len(atts))
-	keys := make([][]byte, len(atts))
-	for i := 0; i < len(atts); i++ {
-		enc, err := proto.Marshal(atts[i])
-		if err != nil {
-			return err
-		}
-		key, err := ssz.HashTreeRoot(atts[i].Data)
-		if err != nil {
-			return err
-		}
-		encodedValues[i] = enc
-		keys[i] = key[:]
-	}
-	return k.db.Batch(func(tx *bolt.Tx) error {
-		bkt := tx.Bucket(attestationsBucket)
-		for i := 0; i < len(atts); i++ {
-			indicesByBucket := createAttestationIndicesFromData(atts[i].Data, tx)
-			if err := updateValueForIndices(indicesByBucket, keys[i], tx); err != nil {
-				return errors.Wrap(err, "could not update DB indices")
-			}
-			if err := bkt.Put(keys[i], encodedValues[i]); err != nil {
-				return err
+	var wg sync.WaitGroup
+	var err error
+	wg.Add(len(atts))
+	for _, a := range atts {
+		go func(w *sync.WaitGroup, att *ethpb.Attestation) {
+			defer wg.Done()
+			if routineErr := k.SaveAttestation(ctx, att); routineErr != nil {
+				err = routineErr
```

Comment:
This is better, but you could have errors overwriting each other in the case there are 2 or more distinct errors. 

What do you think about a list of errors, then concatenating at the end?